x-pack/plugins/infra/server/plugin.ts

```diff
@@ -118,6 +121,20 @@ export class InfraServerPlugin implements Plugin<InfraPluginSetup> {
       metrics: new InfraMetricsDomain(new KibanaMetricsAdapter(framework)),
     };
 
+    const logsAlertClient = initializeAlertClient({
+      name: 'logs',
+      getStartServices: core.getStartServices,
```

Comment:
We only need to pass this so the initializer can create a default client which just defaults to the internalUser client. I think the ruleRegistry plugin could default to that and do its own grabbing from startServices, and we could allow for the ability to specify a client here, which would be more semantically pleasing imo.